kliff/models/kim.py

```diff
@@ -531,39 +515,19 @@ def create_a_kim_compute_argument(self):
 
         return kim_ca
 
-    def set_opt_params(self, **kwargs):
+    def set_params_mutable(self, list_of_params: List[str]):
         """
-        Set the parameters that will be optimized.
-
-        One or more parameters can be set. Each argument is for one parameter, where the
-        argument name is the parameter name, the value of the argument is the
-        settings(including initial value, fix flag, lower bound, and upper bound).
-
-        The value of the argument should be a list of list, where each inner list is for
-        one component of the parameter, which can contain 1, 2, or 3 elements.
-         See `~kliff.model.model.Model.read_opt_params()` for the options of the elements.
-
+        Set all the optimizable parameters from list of names of parameters
+        Args:
+            list_of_params: List of string names of parameters
         Example:
-           instance.set(A=[['DEFAULT'], [2.0, 1.0, 3.0]], B=[[1.0, 'FIX'], [2.0, 'INF', 3.0]])
+            model.set_params_mutable(["A", "B", "sigma"])
         """
-        self.opt_params.set(**kwargs)
-
-        # update kim internal model param (note, set_one will update model_params)
-        for name, _ in kwargs.items():
-            p_idx = self.model_params[name].index
-            for c_idx, v in enumerate(self.model_params[name].value):
-                try:
-                    self.kim_model.set_parameter(p_idx, c_idx, v)
-                except RuntimeError:
-                    raise kimpy.KimPyError(
-                        "Calling `kim_model.set_parameter()` failed."
-                    )
-
-        try:
-            self.kim_model.clear_then_refresh()
-        except RuntimeError:
-            raise kimpy.KimPyError("Calling `kim_model.clear_then_refresh()` failed.")
-
+        for param in list_of_params:
+            self.model_params[param].opt_mask = np.ones_like(
```

Comment:
Let's call `set_opt_mask` instead of modifying the attributes directly?

Reply:
Current function is named `add_opt_mask`. I have modified this to call `add_opt_mask` let me know if we should rename it to `set_opt_mask`. `add_opt_mask` is similar to other function names `add_transform` etc. I can name the `set_*` and have no preference one way or another.